### PR TITLE
TXAD-19: Add permit user interface

### PR DESCRIPTION
**Pull request by objectorientguy**

### CODE REVIEW

File: lib/screens/home/widgets/online_modules.dart [hunk 1]
```diff
@@ -59,6 +60,9 @@ class OnLineModules extends StatelessWidget {
 
   navigateToModule(index, context) {
     switch (index) {
+      case 4:
+        Navigator.pushNamed(context, PermitScreen.routeName);
```
Comment (Aditi-005) [inline lib/screens/home/widgets/online_modules.dart:64]: Why not PermitListScreen?

File: lib/screens/permit/permit_details_screen.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              child: ListTile(
+                title: Padding(
+                  padding: const EdgeInsets.only(top: midTiniestSpacing),
+                  child: Row(
+                    crossAxisAlignment: CrossAxisAlignment.center,
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      const Text('WP - 00197 (!)'),
+                      Container(
+                        width: MediaQuery.of(context).size.width * 0.25,
```
Comment (Aditi-005) [inline lib/screens/permit/permit_details_screen.dart:45]: why not defined in app dimensions

File: lib/screens/permit/permit_details_screen.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                subtitle: const Padding(
+                  padding: EdgeInsets.only(top: midTiniestSpacing),
+                  child: Tags(),
+                ),
+              ),
+            ),
+            const SizedBox(height: midTiniestSpacing),
+            const Divider(
+              height: kDividerHeight,
+              thickness: 1,
```
Comment (Aditi-005) [inline lib/screens/permit/permit_details_screen.dart:70]: why not in app dimensions

File: lib/screens/permit/permit_screen.dart [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                            Navigator.pushNamed(
+                                context, PermitDetailsScreen.routeName);
+                          },
+                          title: Row(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                            children: [
+                              const Text('WP - 00197 (!)'),
+                              Container(
+                                width: MediaQuery.of(context).size.width * 0.25,
```
Comment (Aditi-005) [inline lib/screens/permit/permit_screen.dart:52]: why not defined in app dimensions

File: lib/screens/permit/permit_screen.dart [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                                height:
+                                    MediaQuery.of(context).size.width * 0.050,
+                                decoration: BoxDecoration(
+                                  color: Colors.lightBlue,
+                                  borderRadius: BorderRadius.circular(5),
```
Comment (Aditi-005) [inline lib/screens/permit/permit_screen.dart:57]: why not defined in app dimensions

File: lib/screens/permit/widgets/permit_group.dart [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                leading: Image.asset(
+                  leadingAvatarIcon,
+                ),
+                title: Row(
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    const Text('Martin Smith'),
+                    Container(
+                      width: MediaQuery.of(context).size.width * 0.15,
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/permit_group.dart:35]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.center,
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: [
+        Container(
+          decoration: BoxDecoration(
+            color: Colors.redAccent,
+            borderRadius: BorderRadius.circular(5),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:15]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+          ),
+          margin: const EdgeInsets.only(right: 5, bottom: 5),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:17]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+          alignment: Alignment.center,
+          child: const Padding(
+            padding: EdgeInsets.all(8.0),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:20]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+              'Expired',
+              style: TextStyle(color: Colors.white),
+              textAlign: TextAlign.center,
+            ),
+          ),
+        ),
+        Container(
+          decoration: BoxDecoration(
+            color: Colors.amberAccent,
+            borderRadius: BorderRadius.circular(5),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:31]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+          ),
+          alignment: Alignment.center,
+          margin: const EdgeInsets.only(right: 5, bottom: 5),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:50]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+          child: const Padding(
+            padding: EdgeInsets.all(8.0),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:36]: why not defined in app dimensions

File: lib/screens/permit/widgets/tags.dart [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+              'NPI',
+              style: TextStyle(color: Colors.white),
+              textAlign: TextAlign.center,
+            ),
+          ),
+        ),
+        Container(
+          decoration: BoxDecoration(
+            color: Colors.amberAccent,
+            borderRadius: BorderRadius.circular(5),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/tags.dart:47]: why not defined in app dimensions

**Review by Aditi-005 — CHANGES_REQUESTED** [review]
Please make the requested changes

File: lib/screens/permit/widgets/tags.dart (hunk 13 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/tags.dart:47]: Done

File: lib/screens/permit/widgets/tags.dart (hunk 12 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/tags.dart:36]: Done

File: lib/screens/permit/widgets/tags.dart (hunk 11 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/tags.dart:50]: Done

File: lib/screens/permit/widgets/tags.dart (hunk 10 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/tags.dart:31]: Done

File: lib/screens/permit/widgets/tags.dart (hunk 9 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/tags.dart:20]: Done

File: lib/screens/permit/permit_screen.dart (hunk 5 above)
Comment (objectorientguy) [reply lib/screens/permit/permit_screen.dart:57]: As discussed in the daily call. This container contains a dynamic data which cant be defined in app dimensions.
As name defines 'app dimensions' has all the defined variables which are consistently used in the app through out.

File: lib/screens/permit/permit_details_screen.dart (hunk 3 above)
Comment (objectorientguy) [reply lib/screens/permit/permit_details_screen.dart:70]: Done

File: lib/screens/home/widgets/online_modules.dart (hunk 1 above)
Comment (objectorientguy) [reply lib/screens/home/widgets/online_modules.dart:64]: Its the default screen of that module, so wanted to keep it this way. 
But yeah, can make it PermitListScreen.

File: lib/screens/permit/permit_details_screen.dart (hunk 2 above)
Comment (Aditi-005) [reply lib/screens/permit/permit_details_screen.dart:48]: Ok

File: lib/screens/permit/widgets/permit_timeline.dart [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+            const SizedBox(width: tiniestSpacing),
+            const Text('24.05.2023')
+          ],
+        ),
+        const SizedBox(height: midTiniestSpacing),
+        Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            Image.asset('assets/icons/clock.png', height: 15, width: 15),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/permit_timeline.dart:75]: Add height and width in app dimensions

File: lib/screens/permit/widgets/permit_timeline.dart [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+  Widget _buildDateAndTime(int index) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: [
+        Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            Image.asset('assets/icons/calendar.png', height: 15, width: 15),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/permit_timeline.dart:65]: Add height and width in app dimensions

**Review by Aditi-005 — CHANGES_REQUESTED** [review]
Please make the changes

**Review by Aditi-005 — APPROVED** [review]
Approved

**Pull request merged**